### ISSUE TITLE
Multiple notifications

### ISSUE THREAD

**Issue by tpzltn** [filipnet/checkmk-telegram-notify#1]
Hi,

Script is working. I have created a bot and a group. So I moved bot to this group and added token+group id to the conf file.

My issue is, I got every notification in the group 3 times. All notifications are coming in 3 times one after one.

Do you have any idea what should cause this?

Thanks.

**Comment by filipnet** [comment]
Hi,

I use the script as it is available here with Check_MK RAW 1.6.0p18, get the messages only once. Telegram group is a "supergroup".
You could try to exclude where the cause of the duplicates is by testing, if you have not already done so.

- If you are also notified by e-mail, validate they only arrive once in your mailbox
- Direct test on the command line with curl (see line 53) e.g. (before of course adjust bot-id and chat-id) and check that this message arrives only once.
`curl -s -X POST "https://api.telegram.org/bot1001001001:AABBCCdd/sendMessage" -d chat_id=-1001001001001 -d text="Test"`

I had duplicates in Telegram with Node-RED, here the cause were problems with the webhook. The status code 200 must always be returned here. Maybe you run into a timeout, you could try it in the string above with `curl -s --max-time 30 ...`. This might help you with your problem.

**Comment by tpzltn** [comment]
Hi,

The test was successful. The problem was that "Notify all users" was checked and I have 3 users in the Cmk. Thank you.

**Comment by filipnet** [comment]
You're welcome, glad it was only a configuration issue and the module works for you. If you like it, don't forget to subscribe and support the repository-rating with a star.
